doc/developer-guides/hld/hld-splitlock.rst

```diff
@@ -0,0 +1,122 @@
+
+- The 29th bit of TEST_CTL MSR(0x33) control enable/disable #AC for Split-locked
+  Access.
+
+Note that P6 family CPUs also has TEST_CTRL MSR, not enumerated in this way and
+don't support #AC for Split-locked Access.
+
+Acrn Handling Split-locked Access
+*********************************
+Split-locked Access is not expected in Acrn hypervisor, should never happen. But
+that is not guaranteed inside VMs. Acrn support use this CPU capability, that
+are design principles:
+
+- Always enable #AC on Split-locked Access in physical side.
+
+- Present virtual split lock capability to guest, and directly deliver #AC to
+  guest (virtual split-lock capability helps the guest to isolate violations
+  from user land).
+
+- Guest write of MSR_TEST_CTL is ignored, guest read gets the written value.
+
+- If Split-locked Access happens in ACRN, this is a software bug we must fix it.
+
+- If split-locked Access happens in guest kernel, the guest may not be able to
+  fix the issue gracefully (the guest may behavior differently with that of
+  native OS). The RT guest should avoid the Split-locked Access, otherwise it is
+  a software bug.
+
+Fore Enable This Feature
+========================
+This feature is enumerated at PCPU pre-initialization stage, where Acrn detects
+ CPU capabilities. If the PCPU supports this feature:
+
+- Enabled at each PCPU post-initialization stage.
+
+- Acrn hypervisor presents virtual emulated TEST_CTRL MSR to each VCPU.
+  Setting/clearing TEST_CTRL[bit 29] from VCPU, has no any effect.
+
+If PCPU does not have this capability, VCPU does not have the virtual TEST_CTRL
+neither.
+
+Expected Behavior in Acrn
+=========================
+Acrn HV should never trigger Split-locked Access, it is not allowed to run with
+Split-locked Access. If it does trigger split-lock, Acrn report #AC at the
+instruction and stop running. This HV instruction is considered as a bug, must
+fix it.
+
+Expected Behavior in VM
+=======================
+If VM process have a Split-locked Access at user space, it will be terminated by
+SIGBUS. When debug inside VM for a reason, you may find it triggers an #AC, no
+matter alignment checking is enabled or not.
+If VM kernel have a Split-locked Access, it would hang or oops for #AC. VM
+kernel may try disable #AC for Split-locked Access and continue, but it will not
+make it. For that case, you may find HV warning message that says VM tries
+writing to TEST_CTRL MSR, and help to identify the problem.
+
+
+Disable Split-locked Access Detection
+=====================================
+When the CPU support Split-locked Access detection, Acrn HV will use it to
+prevent any VM to run a burden software(which contains split-locked instructions). Sometimes customer may really want to run the software and does not care
+about performance of entire system, he can configure this feature out of Acrn
+code.
+To use Acrn Configuration Tools to disable split lock detection, that feature
```

Comment:
Which feature is not finished yet, the feature itself or the ability to disable it via the config tool?

Reply:
Will fix that ambiguity , not finish disable it via config tool.